firefox-ios/Client/Application/BackgroundFirefoxSuggestIngestUtility.swift

```diff
@@ -40,6 +41,7 @@ class BackgroundFirefoxSuggestIngestUtility: BackgroundUtilityProtocol, FeatureF
 
     /// Submits a request to schedule the background ingestion task.
     private func submitBackgroundTaskRequest() throws {
+        guard didRegisterTaskHandler else { return }
```

Comment:
The background task handler also calls `submitBackgroundTaskRequest()` to retry if ingestion fails. Is that allowed, and can we expect that `didRegisterTaskHandler` will be set correctly there?

(I think the answer to both is "yes", if I'm understanding how `BackgroundFetchAndProcessingUtility` and `BGTaskScheduler` work? But I wanted to double-check with you).

Reply:
> The background task handler also calls submitBackgroundTaskRequest() to retry if ingestion fails. Is that allowed, and can we expect that didRegisterTaskHandler will be set correctly there?

AFAICS the `submitBackgroundTaskRequest()` is only called in two places. One of them is within the Task block for the task registration itself, so that should always be fine since we know the setup has been performed in that case (and the related state should be set before the block is ever run, AFAICT).

The other callsite is `scheduleTaskOnAppBackground()` which is the problematic one, since I believe that can be triggered in some scenarios where `setUp()` hasn't actually been called yet. 

But if anything with the above doesn't sound correct LMK. I'm not super-familiar with this code so if you think there might be a scenario that is overlooked I'm happy to dig into it further.

Reply:
That all makes sense to me, too—thanks!